subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/resolver/MavenMetadataLoader.java

```diff
@@ -53,8 +59,14 @@ public MavenMetadata load(URI metadataLocation) throws ResourceException {
         return metadata;
     }
 
-    private void parseMavenMetadataInfo(final URI metadataLocation, final MavenMetadata metadata) {
-        ExternalResource resource = repository.getResource(metadataLocation, true);
+    private void parseMavenMetadataInfo(final URI metadataLocation, final MavenMetadata metadata) throws IOException {
+        ExternalResource resource = cacheAwareExternalResourceAccessor.getResource(metadataLocation, new CacheAwareExternalResourceAccessor.ResourceFileStore() {
```

Comment:
I read your comment. This has been introduced after a discussion with @bigdaz , seeing that we did quite a few requests for metadata. The drawback of this approach is that if we need to revalidate the metadata file, that is has effectively changed but was still small, then we would _maybe_ be slower. One option would be to decide whether to do a `HEAD` request or not based on the size of the document that is in cache. It would be independent of whether we fetch maven metadata or something else, just a heuristic done by the accessor. I wouldn't make this part of this story, though.